README.md
# RiProG-AI

![RiProG-AI Banner](https://raw.githubusercontent.com/RiProG-id/RiProG-AI/main/banner.jpg)

**ALL IN ONE Magisk Module**

Enhance the Android experience with a combination of tweaks that automatically adapt mode and provide a description based on the latest version.

**RiProG Open Source** | [Channel](https://t.me/RiOpSo) | [Discussion](https://t.me/RiOpSoDisc)

## Key Features

- Disable Kernel Panic
- Disable printk
- Disable Various Debugger
- Disable Various Log
- Fuse Passthrough
- Low Memory Consumption
- Modifying Scheduler Features
- SurfaceFlinger Improvement
- Virtual Memory Optimization
- HWUI Cache Optimizer
- Smart I/O Storage
- Smart CPU Optimization
- Smart GPU Optimization
- Smart Low Memory Killer
- Smart Throttling Mitigation

## Changelog

**Version 5.0**

- Fixed broken code.
- Reworked some properties.
- Improved I/O Scheduler.
- Resolved lag issues when changing modes.
- Introduced Smart Throttling Mitigation.
- Implemented a new detection algorithm.
- Fixed AI log not updating.
- Limited CPU in sleep mode.
- Various other bug fixes and improvements.

**Version 5.1**

- Fixed unstable GPU
- Added Media Codex Acceleration
- Added Prevent streaming buffering
- Reduced cpu frequency in sleep mode
- Increased minimum cpu frequency in daily mode

**Version 5.2**

- Code runs alternately
- Remove video hardware acceleration
- Disable surfaceflinger hardware
- Make changes to the code's behavior
- Several bug fixes

**Version 5.3**

- Various bug fixes
- High GPU usage fixes
- Overheating issues fixes
- errors in some code fixes
- Improved efficiency of AI log
- Update the game list filter detector

**Version 5.4**

- Cpufreq calculation errors fixes
- Gamelist reader errors fixes
- Sleep mode errors fixes
- Module runs without using nohup in the background

## Support the Developer

If you appreciate the work done on RiProG-AI, consider supporting the developer:

[![Support RiProG](https://img.shields.io/badge/Support%20Dev-Saweria-brightgreen)](https://saweria.co/RiProG)


changelog.md
# RiProG AI

![RiProG AI Banner](https://raw.githubusercontent.com/RiProG-id/RiProG-AI/main/banner.jpg)

**ALL IN ONE Magisk Module**

Enhance the Android experience with a combination of tweaks. This module automatically adapts mode and provides a description based on the latest version.

**RiProG Open Source** | [Channel](https://t.me/RiOpSo) | [Discussion](https://t.me/RiOpSoDisc)

## Changelog

**Version 5.0**

- Fixed broken code.
- Reworked some properties.
- Improved I/O Scheduler.
- Resolved lag issues when changing modes.
- Introduced Smart Throttling Mitigation.
- Implemented a new detection algorithm.
- Fixed AI log not updating.
- Limited CPU in sleep mode.
- Various other bug fixes and improvements.

**Version 5.1**

- Fixed unstable GPU
- Added Media Codex Acceleration
- Added Prevent streaming buffering
- Reduced cpu frequency in sleep mode
- Increased minimum cpu frequency in daily mode

**Version 5.2**

- Code runs alternately
- Remove video hardware acceleration
- Disable surfaceflinger hardware
- Make changes to the code's behavior
- Several bug fixes

**Version 5.3**

- Various bug fixes
- High GPU usage fixes
- Overheating issues fixes
- errors in some code fixes
- Improved efficiency of AI log
- Update the game list filter detector

**Version 5.4**

- Cpufreq calculation errors fixes
- Gamelist reader errors fixes
- Sleep mode errors fixes
- Module runs without using nohup in the background


install.sh
#!/bin/sh
SKIPMOUNT=false
PROPFILE=false
POSTFSDATA=true
LATESTARTSERVICE=true
REPLACE="

"
sleep 2
ui_print ""
ui_print "************************************"
ui_print "*    RiProG Open Source @ RiOpSo   *"
ui_print "************************************"
ui_print "*      Muhammad Rizki @ RiProG     *"
ui_print "************************************"
ui_print ""
sleep 2
ui_print "- Extracting module files"
unzip -o "$ZIPFILE" 'gamelist.txt' -d "$MODPATH" >&2
mkdir -p "$MODPATH/system/bin"
unzip -o "$ZIPFILE" 'AI' -d "$MODPATH/system/bin" >&2
ui_print "- Detecting installed game"
sleep 2
counter=1
package_list=$(cmd package list packages | cut -f 2 -d ":")
game_list=$(cat "$MODPATH/gamelist.txt")
echo "$game_list" | while IFS= read -r gamelist; do
    line=$(echo "$gamelist" | awk '!/ /')
    if echo "$package_list" | grep -q "$line"; then
        ui_print "  $counter. $line"
        counter=$((counter + 1))
    else
        sed -i "/$line/d" "$MODPATH/gamelist.txt"
    fi
done
ui_print "  If your game is not detected"
ui_print "  Edit gamelist.txt, add, and reinstall"
sleep 2
chmod +x "$MODPATH/system/bin/AI"

common/post-fs-data.sh
#!/bin/sh
android_properties="
persist.sys.dalvik.hyperthreading=true
persist.sys.dalvik.multithread=true
ro.hwui.texture_cache_size=128
ro.hwui.layer_cache_size=64
ro.hwui.r_buffer_cache_size=16
ro.hwui.path_cache_size=64
ro.hwui.gradient_cache_size=2
ro.hwui.drop_shadow_cache_size=8
rro.hwui.texture_cache_flushrate=0.2
ro.hwui.text_small_cache_width=2048
ro.hwui.text_small_cache_height=2048
ro.hwui.text_large_cache_width=4096
ro.hwui.text_large_cache_height=4096
ro.iorapd.enable=false
iorapd.perfetto.enable=false
iorapd.readahead.enable=false
persist.device_config.runtime_native_boot.iorap_readahead_enable=false
persist.sys.purgeable_assets=1
ro.media.dec.aud.flac.enabled=1
ro.media.dec.aud.wma.enabled=1
ro.media.dec.vid.avi.enabled=1
ro.media.dec.vid.wmv.enabled=1
ro.media.dec.aud.wma.enabled=1
ro.media.dec.aud.mp3.enabled=1
ro.media.dec.vid.mp4.enabled=1
ro.media.dec.aud.flac.enabled=1
ro.media.dec.vid.H.264.enabled=1
ro.media.enc.aud.flac.enabled=1
ro.media.enc.vid.H.264.enabled=1
ro.media.enc.aud.wma.enabled=1
ro.media.enc.vid.wmv.enabled=1
ro.media.enc.aud.mp3.enabled=1
ro.media.enc.vid.mp4.enabled=1
media.stagefright.enable-player=true
media.stagefright.enable-meta=true
media.stagefright.enable-scan=true
media.stagefright.enable-http=true
media.stagefright.enable-rtsp=true
media.stagefright.enable-record=false
"
echo "$android_properties" | while IFS= read -r prop; do
    prop_name="${prop%%=*}"
    prop_value="${prop#*=}"
    setprop"$prop_name" "$prop_value"
done


common/service.sh
#!/bin/sh
read_file() {
  if [ -f "$1" ]; then
    if [ ! -r "$1" ]; then
      chmod +r "$1"
    fi
    cat "$1"
  fi
}
write_file() {
  if [ -f "$1" ]; then
    if [ ! -w "$1" ]; then
      chmod +w "$1"
    fi
    echo "$2" >"$1"
  fi
}
sleep 1
while [ -z "$(resetprop sys.boot_completed)" ]; do
  sleep 5
done
sleep 1
write_file /proc/sys/kernel/panic "0"
write_file /proc/sys/kernel/panic_on_oops "0"
write_file /proc/sys/kernel/panic_on_rcu_stall "0"
write_file /proc/sys/kernel/panic_on_warn "0"
write_file /sys/module/kernel/parameters/panic "0"
write_file /sys/module/kernel/parameters/panic_on_warn "0"
write_file /sys/module/kernel/parameters/panic_on_oops "0"
write_file /proc/sys/kernel/printk "0 0 0 0"
write_file /proc/sys/kernel/printk_devkmsg "off"
write_file /sys/module/printk/parameters/console_suspend "Y"
write_file /sys/module/printk/parameters/cpu "N"
write_file /sys/module/printk/parameters/ignore_loglevel "Y"
write_file /sys/module/printk/parameters/pid "N"
write_file /sys/module/printk/parameters/time "N"
write_file /sys/kernel/printk_mode/printk_mode "0"
sleep 1
find /sys/ -name "*log*" -o -name "*debug*" | while IFS= read -r logdebug; do
  write_file "$logdebug" "0"
done
sleep 1
write_file /proc/sys/kernel/sched_tunable_scaling "1"
write_file NO_GENTLE_FAIR_SLEEPERS:1 >/sys/kernel/debug/sched_features
write_file START_DEBIT:1 >/sys/kernel/debug/sched_features
write_file NEXT_BUDDY:1 >/sys/kernel/debug/sched_features
write_file LAST_BUDDY:1 >/sys/kernel/debug/sched_features
write_file STRICT_SKIP_BUDDY:1 >/sys/kernel/debug/sched_features
write_file CACHE_HOT_BUDDY:1 >/sys/kernel/debug/sched_features
write_file WAKEUP_PREEMPTION:1 >/sys/kernel/debug/sched_features
write_file NO_HRTICK:1 >/sys/kernel/debug/sched_features
write_file NO_DOUBLE_TICK:1 >/sys/kernel/debug/sched_features
write_file LB_BIAS:1 >/sys/kernel/debug/sched_features
write_file NONTASK_CAPACITY:1 >/sys/kernel/debug/sched_features
write_file NO_TTWU_QUEUE:1 >/sys/kernel/debug/sched_features
write_file NO_SIS_AVG_CPU:1 >/sys/kernel/debug/sched_features
write_file RT_PUSH_IPI:1 >/sys/kernel/debug/sched_features
write_file NO_FORCE_SD_OVERLAP:1 >/sys/kernel/debug/sched_features
write_file NO_RT_RUNTIME_SHARE:1 >/sys/kernel/debug/sched_features
write_file NO_LB_MIN:1 >/sys/kernel/debug/sched_features
write_file ATTACH_AGE_LOAD:1 >/sys/kernel/debug/sched_features
write_file ENERGY_AWARE:1 >/sys/kernel/debug/sched_features
write_file NO_MIN_CAPACITY_CAPPING:1 >/sys/kernel/debug/sched_features
write_file NO_FBT_STRICT_ORDER:1 >/sys/kernel/debug/sched_features
write_file EAS_USE_NEED_IDLE:1 >/sys/kernel/debug/sched_features
sleep 1
write_file /proc/sys/vm/drop_caches "3"
write_file /proc/sys/vm/dirty_background_ratio "20"
write_file /proc/sys/vm/dirty_expire_centisecs "1000"
write_file /proc/sys/vm/page-cluster "0"
write_file /proc/sys/vm/dirty_ratio "10"
write_file /proc/sys/vm/laptop_mode "0"
write_file /proc/sys/vm/block_dump "1"
write_file /proc/sys/vm/compact_memory "0"
write_file /proc/sys/vm/dirty_writeback_centisecs "5000"
write_file /proc/sys/vm/oom_dump_tasks "1"
write_file /proc/sys/vm/oom_kill_allocating_task "0"
write_file /proc/sys/vm/stat_interval "60"
write_file /proc/sys/vm/panic_on_oom "0"
write_file /proc/sys/vm/swappiness "20"
write_file /proc/sys/vm/vfs_cache_pressure "50"
write_file /proc/sys/vm/overcommit_ratio "80"
write_file /proc/sys/vm/extra_free_kbytes "10240"
write_file /proc/sys/kernel/random/read_wakeup_threshold "64"
write_file /proc/sys/kernel/random/write_wakeup_threshold "128"
sleep 1
android_properties="
debug.sf.disable_backpressure=1
debug.sf.latch_unsignaled=1
debug.sf.enable_hwc_vds=0
debug.sf.early_phase_offset_ns=500000
debug.sf.early_app_phase_offset_ns=500000
debug.sf.early_gl_phase_offset_ns=3000000
debug.sf.early_gl_app_phase_offset_ns=15000000
debug.sf.high_fps_early_phase_offset_ns=6100000
debug.sf.high_fps_early_gl_phase_offset_ns=650000
debug.sf.high_fps_late_app_phase_offset_ns=100000
debug.sf.phase_offset_threshold_for_next_vsync_ns=6100000
debug.sf.showupdates=0
debug.sf.showcpu=0
debug.sf.showbackground=0
debug.sf.showfps=0
debug.sf.hw=0
"
echo "$android_properties" | while IFS= read -r prop; do
  prop_name="${prop%%=*}"
  prop_value="${prop#*=}"
  resetprop -n "$prop_name" "$prop_value"
done
sleep 1
AI >/dev/null 2>&1
